src/mitsuba/mitsuba.cpp

```diff
@@ -223,8 +219,8 @@ int main(int argc, char *argv[]) {
 
         // Initialize Intel Thread Building Blocks with the requested number of threads
         if (*arg_threads)
-            __global_thread_count = arg_threads->as_int();
-        tbb::task_scheduler_init scheduler(std::max(1, (int) __global_thread_count));
+            Thread::set_thread_count(std::max(1, arg_threads->as_int()));
+        jit_llvm_set_thread_count(Thread::thread_count());
```

Comment:
Is this perhaps unnecessary? My guess would be taht both of those methods will update the enoki-jit thread pool size, no?

Reply:
I will clean up that logic